app/controllers/course/material/folders_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Course::Material::FoldersController < Course::ComponentController
+  load_and_authorize_resource :material_folder, through: :course, parent: false,
```

Comment:
Sorry, I didn't get you.  Do you mean we should use `folders` since it be also owned by others ?